src/test/java/org/opentripplanner/middleware/ApiUserFlowTest.java

```diff
@@ -0,0 +1,169 @@
+import static org.opentripplanner.middleware.auth.Auth0Connection.authDisabled;
+import static org.opentripplanner.middleware.auth.Auth0Users.createAuth0UserForEmail;
+import static org.opentripplanner.middleware.controllers.api.ApiUserController.DEFAULT_USAGE_PLAN_ID;
+
+/**
+ * Tests to simulate API user flow. The following config parameters must be set in configurations/default/env.yml for
+ * these end-to-end tests to run: - AUTH0_DOMAIN set to a valid Auth0 domain - AUTH0_API_CLIENT set to a valid Auth0
+ * application client id - AUTH0_API_SECRET set to a valid Auth0 application client secret - DEFAULT_USAGE_PLAN_ID set
+ * to a valid usage plan id (AWS requires this to create an api key). - An AWS_PROFILE is required, or AWS access has
+ * been configured for your operating environment e.g. C:\Users\<username>\.aws\credentials in Windows or Mac OS
+ * equivalent.
+ *
+ * The following environment variable must be set for these tests to run: - RUN_E2E=true
+ *
+ * Auth0 must be correctly configured as described here: https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow
+ */
+public class ApiUserFlowTest {
+    private static ApiUser apiUser;
+    private static OtpUser otpUser;
+
+    /**
+     * Create an {@link ApiUser} and an {@link AdminUser} prior to unit tests
+     */
+    @BeforeAll
+    public static void setUp() throws IOException {
+        OtpMiddlewareTest.setUp();
+        // As a pre-condition, create an API User with API key.
+        apiUser = PersistenceUtil.createApiUser(String.format("test-%s@example.com", UUID.randomUUID().toString()));
+        apiUser.createApiKey(DEFAULT_USAGE_PLAN_ID, true);
+
+        otpUser = new OtpUser();
+        otpUser.email = String.format("test-%s@example.com", UUID.randomUUID().toString());
+        otpUser.hasConsentedToTerms = true;
+        otpUser.storeTripHistory = true;
+
+        if (!authDisabled()) {
+            // create Auth0 users for apiUser and optUser.
+            try {
+                User auth0User = createAuth0UserForEmail(apiUser.email, TestUtils.TEMP_AUTH0_USER_PASSWORD);
+                // update api user with valid auth0 user ID (so the Auth0 delete works)
+                apiUser.auth0UserId = auth0User.getId();
+                Persistence.apiUsers.replace(apiUser.id, apiUser);
+
+                auth0User = createAuth0UserForEmail(otpUser.email, TestUtils.TEMP_AUTH0_USER_PASSWORD);
+                // update otp user with valid auth0 user ID (so the Auth0 delete works)
+                otpUser.auth0UserId = auth0User.getId();
+            } catch (Auth0Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * Delete the users if they were not already deleted during the test script.
+     */
+    @AfterAll
+    public static void tearDown() {
+        apiUser = Persistence.apiUsers.getById(apiUser.id);
```

Comment:
Needs a `assumeTrue(getBooleanEnvVar("RUN_E2E"));` on the first line of this method so that this only runs during E2E testing.

Reply:
Updated.